echopop/workflow/feat_hake.py

```diff
@@ -163,124 +168,482 @@
+kriging_params_dict, variogram_params_dict = load_data.load_kriging_variogram_params(
+    geostatistic_params_filepath,
+    geostatistic_params_sheet_name,
+    FEAT_TO_ECHOPOP_GEOSTATS_PARAMS_COLUMNS
+)
+
+# ==================================================================================================
+# ==================================================================================================
+# DATA PROCESSING
+# ==================================================================================================
+# Generate binned distributions [age, length]
+# -------------------------------------------
+age_bins: npt.NDArray[np.number] = np.linspace(start=1., stop=22., num=22)
+length_bins: npt.NDArray[np.number] = np.linspace(start=2., stop=80., num=40)
+
+# Outputs:
+# ---- `pandas.DataFrame` that includes the `pandas`-formatted intervals used for cutting data
+# ---- NOTE: This function has an argument `return_dataframe` that will return a Tuple rather than 
+# ---- a `pandas.DataFrame` when set to `False`. This was primarily for debugging purposes, so it 
+# ---- can be removed if it ends up never being used
+age_distribution: pd.DataFrame = utils.binned_distribution(age_bins)
+length_distribution: pd.DataFrame = utils.binned_distribution(length_bins)
 
-root_path = "WHERE_ALL_DATA_ARE"
-species_code = "SPECIES_CODE"
-df_nasc_no_age1: pd.DataFrame  # extracted nasc data from above, can also be df_nasc_all_ages
+# ==================================================================================================
+# 'Binify' the biological data [age, length]
+# ------------------------------------------
+# Note: this argument can be done 'inplace'
+data: Union[pd.DataFrame, Dict[str, pd.DataFrame]] = dict_df_bio_ks 
+bin_distribution: pd.DataFrame = length_distribution.copy()
+bin_column: str = "length"
+inplace: bool = False
 
-bio_path_dict: dict  # the "biological" section of year_config.yml
-# this will be simplified now that we read from the master spreadsheet
-strata_path_dict: dict  # the "stratification" section of year_config.yml
+# Outputs:
+# ---- `Union[pd.DataFrame, Dict[str, pd.DataFrame], None]`
+# ---- `pandas.DataFrame` if a single `pandas.DataFrame` input
+# ---- The same structured `Dict[str, pandas.DataFrame]` if a `Dict[str, pandas.DataFrame]` input
+# ---- `None` when `inplace=True`
 
-dict_df_bio = load_data.load_biological_data(root_path, bio_path_dict, species_code)
-dict_df_strata = load_data.load_stratification(root_path, strata_path_dict)
+# Length
+dict_df_bio_binned_ks: Dict[str, pd.DataFrame] = utils.binify(
+    data, bin_distribution, bin_column, inplace
+)
 
-# Consolidate all input data into df_acoustic_dict
-df_nasc_no_age1 = load_data.consolidate_all_data(
-    df_nasc=df_nasc_no_age1, df_bio_dict=dict_df_bio, df_strata_dict=dict_df_strata
+# Age
+utils.binify(
+    data=dict_df_bio_binned_ks, bin_distribution=age_distribution, bin_column="age", inplace=True
 )
```

Comment:
I think it's better to have the functions just do the binning in place and not having it as an option, to reduce confusion.  I.e. already have inplace=True.

Reply:
Also make the function only operate on a dataframe and not a dictionary to reduce confusion.
Expose the dictionary structure by doing a dictionary comprehension and call the function within it.